packages/profile-sync-controller/src/shared/encryption/encryption.ts

```diff
@@ -171,6 +171,32 @@ class EncryptorDecryptor {
     return bytesToUtf8(this.#decrypt(ciphertextAndNonce, key));
   }
 
+  getSalt(encryptedDataStr: string) {
+    try {
+      const encryptedData: EncryptedPayload = JSON.parse(encryptedDataStr);
+      if (encryptedData.v === '1') {
+        if (encryptedData.t === 'scrypt') {
+          const { d: base64CiphertextAndNonceAndSalt, saltLen } = encryptedData;
```

Comment:
Optional Perf - can we create a separate method called getSaltLength()?

This method does not need to perform the `base64ToByteArray` conversion, and data slicing. Instead it can rely and return the `saltLen`.

Reply:
Main reason I'm bringing this up is that Mobile/React Native has burned us. Because the Hermes engine does not have JIT, performing JS based operations repeatedly can hurt.

(also why I am worried about the AES-GCM JS implementation, this may need to be swapped with a native solution if we find perf issues)

Reply:
This is optional - so I don't want it blocking this PR :)

Reply:
OK we cannot do the `getSaltLen` approach as we have now opted for a shared salt.

I am closing this discussion, but keep in mind the AES-GCM JS performance on mobile. This has not been thoroughly tested.